GetIntoTeachingApiTests/Services/StoreTests.cs

```diff
@@ -495,6 +495,36 @@ public async void GetUpcomingTeachingEventsByTypeAsync_ReturnsUpcomingEventsGrou
             dates.Should().OnlyContain(d => d >= DateTime.UtcNow);
         }
 
+        [Fact]
+        public async Task GetNumberOfUnknownLocations_ReturnsUnknownLocationsCount()
+        {
+            var locations = new List<Location>() {
+                new Location { Postcode =  "ky119yu", Source = Source.Unknown },
+                new Location { Postcode = "ca48le", Source = Source.CSV },
+            };
+            DbContext.Locations.AddRange(locations);
+            await DbContext.SaveChangesAsync();
+
+            int result = _store.GetNumberOfUnknownLocations();
+
+            result.Should().Be(1);
+        }
+
+        [Fact]
+        public async Task RemoveUnknownLocations_RemovesUnknownLocations()
+        {
+            var locations = new List<Location>() {
+                new Location { Postcode =  "ky119yu", Source = Source.Unknown },
+                new Location { Postcode = "ca48le", Source = Source.CSV },
+            };
+            DbContext.Locations.AddRange(locations);
+            await DbContext.SaveChangesAsync();
+
+            await _store.RemoveUnknownLocations();
+
+            DbContext.Locations.Count().Should().Be(1);

```

Comment:
Sorry I missed this on my first review, but it would be better to have an expectation that none of the `DbContext.Locations` have `Source.Unknown` - you could update the code to remove all `CSV` type locations and this test would still pass.

```
DbContext.Locations.All(l => l.Source != Source.Unknown).Should().BeTrue();
```